### PR TITLE
fix zlip, bzip2 and freetype building. other misc bugs

### PR DESCRIPTION
**Pull request by ITotalJustice**
upgrades:
zlib 1.2.11 -> 1.2.13
libpng 1.5.10 -> 1.5.30
bzip2 1.0.6 -> 1.0.8
freetype 2.10.4 -> 2.12.1

fixes:
- updated broke zlib url
- bzip2 was detecting system installed libs (harfbuzz and brotli) so i disabled them
- add back ntfs lib
- iso9660.c was checking if a bool was negative...
- iso9660.c was using strnicmp instead of strncasecmp

### CODE REVIEW

**Comment by ITotalJustice** [comment]
i found this repo has gcc 9.4 patches <https://github.com/gligli/libxenon/commit/ecfea8fbdc605e0631deedbfb12daadaa3c82c4f>, should i pull this commit into this pr?

File: toolchain/build-xenon-toolchain [hunk 1]
```diff
@@ -302,6 +302,22 @@ function freetype_install
 	echo -e "Done"
 }
 
+# pass the repo name to this function
+function sdl_install
+{
+	echo -e "Cloning $1... "
+	git clone https://github.com/ITotalJustice/$1.git  >> $LOGFILE 2>&1 || fail_with_info
```
Comment (Swizzy) [inline toolchain/build-xenon-toolchain:309]: For this PR to be approved, this needs to not point to your personal repo.

Could you update it to not be your personal repo?

**Comment by Swizzy** [comment]
Unfortunately, the iso9660 change shouldn't be done as it's not a true boolean (0 or 1) it's a integer really, for example it could return -8 which means there's no disc inserted as you can see here; https://github.com/Free60Project/libxenon/blob/master/libxenon/drivers/diskio/ata.c#L571

Technically, this line here is wrong: https://github.com/Free60Project/libxenon/blob/master/libxenon/drivers/diskio/ata.c#L572 since it checks for non-0 as true when infact it should check if it's positive.

File: toolchain/build-xenon-toolchain (hunk 1 above)
Comment (ITotalJustice) [reply toolchain/build-xenon-toolchain:309]: Sure, I'll have a look at this pr again when I'm free. Not sure why I linked my own repo

**Comment by ITotalJustice** [comment]
Hi, i likely won't have time to come back to this pr. Feel free to pick anything from it that's useful

**Pull request closed without merging**